2001-11-24  Sam Hartman  <hartmans@mit.edu>

	* commands.c main.c:  telnet help should telnet to a host called
	help not print an incorrect usage message; patch from  Graeme
	Mathieson <graeme@mathie.cx> 

Kerberos RT #3269

--- krb5-1.4.2.orig/src/appl/telnet/telnet/commands.c
+++ krb5-1.4.2/src/appl/telnet/telnet/commands.c
@@ -2446,8 +2446,6 @@
     cmd = *argv;
     --argc; ++argv;
     while (argc) {
-	if (isprefix(*argv, "help") || isprefix(*argv, "?"))
-	    goto usage;
 	if (strcmp(*argv, "-l") == 0) {
 	    --argc; ++argv;
 	    if (argc == 0)
@@ -2472,8 +2470,7 @@
 	    continue;
 	}
     usage:
-	printf("usage: %s [-l user] [-a] host-name [port]\r\n", cmd);
-	return 0;
+	return 2;
     }
     if (hostp == 0)
 	goto usage;
--- krb5-1.4.2.orig/src/appl/telnet/telnet/main.c
+++ krb5-1.4.2/src/appl/telnet/telnet/main.c
@@ -333,10 +333,14 @@
 
 		if (setjmp(toplevel) != 0)
 			Exit(0);
-		if (tn(argp - args, args) == 1)
-			return (0);
-		else
-			return (1);
+		{
+			int ret = tn(argp - args, args);
+			switch (ret) {
+				case 1: return 0;
+				case 2: usage();
+				default: return 1;
+			}
+		}
 	}
 	(void)setjmp(toplevel);
 	for (;;) {
